### PR TITLE
with the Whatsapp chat window UI

### PR DESCRIPTION
**Pull request by KevinSuvagiya01**
and emoji data update05/12/2024 01:48PM

### CODE REVIEW

File: force-app/main/default/classes/ChatWindowController.cls [hunk 1]
```diff
@@ -1,53 +1,69 @@
...
+    ********************************************************
+    */
     @AuraEnabled
     public static List<Chat__c> fetchAllChats(String contactId){
         List<Chat__c> chats = new List<Chat__c>();
         try {
-            // Your Code
-            chats = [SELECT Id,Type_of_Message__c, Message__c, Time_of_Message__c, Message_Status__c, CreatedDate, Contact__r.FirstName FROM Chat__c WHERE Contact__c =:contactId WITH SECURITY_ENFORCED ORDER BY CreatedDate asc];
-            // Integer result = 10 / 0;
+            chats = [SELECT Id,Type_of_Message__c, Message__c, Time_of_Message__c, Message_Status__c, Is_Text__c, Is_Image__c, Reply_To__c, Message_By__c, Reaction__c, CreatedDate, Contact__r.FirstName FROM Chat__c WHERE Contact__c =:contactId WITH SECURITY_ENFORCED ORDER BY CreatedDate asc];
```
Comment (MRG6804) [inline force-app/main/default/classes/ChatWindowController.cls:30]: Remove unnecessary fields and add logic in the JS to handle the boolean values. using the List of objects and find() a method to update the message content if needed.

File: force-app/main/default/classes/ChatWindowController.cls [hunk 2]
```diff
@@ -1,53 +1,69 @@
...
+            String recordId = chatData.get('recordId');
+            String replyToChatId = chatData.get('replyToChatId');
+
             Chat__c chat = new Chat__c();
             DateTime now = DateTime.now();
-            chat.Message__c = message;
+            chat.Message__c = (messageType == 'Image') ? '/sfc/servlet.shepherd/version/download/'+ message : message;
             chat.Whatsapp_Template__c = templateId;
             chat.Contact__c = recordId;
             chat.Type_of_Message__c = 'Action';
```
Comment (MRG6804) [inline force-app/main/default/classes/ChatWindowController.cls:69]: Remove this

File: force-app/main/default/classes/ChatWindowController.cls [hunk 3]
```diff
@@ -69,7 +101,36 @@ public with sharing class ChatWindowController {
...
+    /*
+    *********************************************************
+    @description     : Method is used to update the reaction of any chat message
+    @param           : chatId - {String} - Id of the chat to update
+    @param           : reaction - {String} - reaction to be updated
+    @return          : chat - {Chat__c} - the updated chat
+    ********************************************************
+    */
+    @AuraEnabled
+    public static Chat__c updateReaction(String chatId, String reaction){
```
Comment (MRG6804) [inline force-app/main/default/classes/ChatWindowController.cls:119]: Change the logic for the Chat reaction as per the discussion.

**Review by MRG6804 — COMMENTED** [review]
Make the changes requested.

File: force-app/main/default/classes/ChatWindowController.cls (hunk 1 above)
Comment (KevinSuvagiya01) [reply force-app/main/default/classes/ChatWindowController.cls]: I have updated this!

File: force-app/main/default/classes/ChatWindowController.cls (hunk 2 above)
Comment (KevinSuvagiya01) [reply force-app/main/default/classes/ChatWindowController.cls]: I have removed this!

File: force-app/main/default/classes/ChatWindowController.cls (hunk 3 above)
Comment (KevinSuvagiya01) [reply force-app/main/default/classes/ChatWindowController.cls:99]: I have changed this as per our discussion!

**Review by MRG6804 — APPROVED** [review]

**Pull request merged**